conbench/entities/benchmark_result.py

```diff
@@ -119,10 +164,10 @@ def create(data):
         if has_error:
             benchmark_result_data["error"] = data["error"]
 
-        # If there was no explicit error *and* the iterations aren't complete, we should add an error
+        # If there was no explicit error *and* at least one iteration failed aren't complete, we should add an error
         if (
             benchmark_result_data.get("error", None) is None
-            and complete_iterations is False
+            and any_errored_iteration is True
```

Comment:
I really don't quite know what we're doing here and why :) https://github.com/conbench/conbench/issues/803

Let's re-think specification, put things into clear text form and then adjust code accordingly. It's OK to break with existing behavior if that will be the result.

Reply:
Cross linking: I added a bunch of text at https://github.com/conbench/conbench/issues/803#issuecomment-1458203148